### DISCUSSION TITLE
How to access the S3 bucket by assume role using camel xml beans.

### DISCUSSION THREAD

**Discussion by ytddqzv** [aws/aws-sdk-java#2926] [General]
access the S3 bucket by assume role using camel xml beans

 

                    AWSCredentials credentials = new BasicAWSCredentials(accessKey, secretKey);

		    ClientConfiguration clientConfig = new ClientConfiguration();
		    clientConfig.setProtocol(Protocol.HTTPS);
		    clientConfig.setProxyHost(protocolProxy);
		    clientConfig.setProxyPort(port);
		    
		    AWSSecurityTokenService stsClient = AWSSecurityTokenServiceClientBuilder
		            .standard()
		            .withRegion("eu-central-1")
		            .withClientConfiguration(clientConfig)
		            .withCredentials(new AWSStaticCredentialsProvider(credentials))
		            .build();

		    AssumeRoleRequest assumeRoleRequest = new AssumeRoleRequest()
		            .withRoleArn(roleArn)
		            .withRoleSessionName(roleSessionName);
		         

		    AssumeRoleResult assumeRoleResult = stsClient.assumeRole(assumeRoleRequest);
		    
		    Credentials sessionCredentials = assumeRoleResult.getCredentials();
		   
		    BasicSessionCredentials awsCredentials = new BasicSessionCredentials(
		    		sessionCredentials.getAccessKeyId(),
		    		sessionCredentials.getSecretAccessKey(),
		            sessionCredentials.getSessionToken());

		    AmazonS3 s3Client= AmazonS3ClientBuilder.standard()
		                            .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
		                            .withClientConfiguration(clientConfig)
		                            .withRegion(Regions.EU_CENTRAL_1)
		                            .build();
			return s3Client;

The above code is written in java language but i need the code in xml beans like below. can any one help/provide the code in xml beans like blow as i am getting issues for the below code.

          <bean class="com.amazonaws.auth.BasicAWSCredentials" id="basicAWSCredentialsBean">
	  <constructor-arg name="accessKey" value="accesskey"/>	
	  <constructor-arg name="secretKey" value="secretkey"/>	
	  </bean>
	
	<bean class="com.amazonaws.ClientConfiguration" id="clientConfiguration">
	    <property name="protocol" ><value type="com.amazonaws.Protocol">${aws.proxy.protocol}</value></property>
        <property name="proxyHost" value="${aws.proxy.host}"/>
        <property name="proxyPort" ><value>${aws.proxy.port}</value></property>
	</bean>
	
	<bean id="awsCredentialsProvider" class="com.amazonaws.auth.AWSStaticCredentialsProvider">
        <constructor-arg ref="basicAWSCredentialsBean"/>
    </bean>
	
	
	<bean id="stsClient" class="com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder" factory-method="standard">
        <property name="region" value="region"/>
        <property name="clientConfiguration" ref="clientConfiguration"/>
        <property name="credentials" ref="awsCredentialsProvider"/>
    </bean>
    
    <bean id="assumeRoleRequestBean" class="com.amazonaws.services.securitytoken.model.AssumeRoleRequest">
      <property name="roleArn" value="roleArn"/>
      <property name="roleSessionName" value="roleSessionName"/>
    </bean>
    
    <bean id="assumeRoleResponseBean" factory-bean="stsClient" factory-method="assumeRole">
        <property name="assumeRoleRequest" ref="assumeRoleRequestBean"/>
    </bean>
 
 	 <bean id="sessionCredentials" class="com.amazonaws.services.securitytoken.model.Credentials"
          factory-bean="assumeRoleResponseBean" factory-method="getCredentials"/>

    <bean id="basicSessionCredentials" class="com.amazonaws.auth.BasicSessionCredentials">
        <constructor-arg name="accessKeyId" value="#{sessionCredentials.accessKeyId}"/>
        <constructor-arg name="secretAccessKey" value="#{sessionCredentials.secretAccessKey}"/>
        <constructor-arg name="sessionToken" value="#{sessionCredentials.sessionToken}"/>
    </bean>

    <bean id="s3Client" class="com.amazonaws.services.s3.AmazonS3ClientBuilder" factory-method="standard">
        <property name="region" value="eu-central-1"/>
        <property name="clientConfiguration" ref="clientConfiguration"/>
        <property name="credentials" ref="basicSessionCredentials"/>
    </bean>

**Comment by debora-ito** [comment]
Moving this to the General topic, as it's not a question about the Java SDK.

**Reply by ytddqzv** [reply]
Actually I want to connect s3 bucket by assume role using xml beans. Could you help on it.
